libs/core/lci_base/src/lci_environment.cpp

```diff
@@ -221,11 +203,9 @@ namespace hpx { namespace util {
 
     std::string lci_environment::get_processor_name()
     {
-        char name[MPI_MAX_PROCESSOR_NAME + 1] = {'\0'};
-        int len = 0;
-        MPI_Get_processor_name(name, &len);
-
-        return name;
+        char hostname[HOST_NAME_MAX + 1] = "Unknown";
+        gethostname(hostname, HOST_NAME_MAX + 1);
+        return hostname;
```

Comment:
Could you please use `asio::ip::host_name();` instead (needs `#include <asio/ip/host_name.hpp>`)?